docs/tasks/deploying-functions.md

```diff
@@ -299,29 +304,25 @@ nuctl deploy \
-can be used to further limit and control communications. The function will be unavailable for invocations
-over HTTP from entities running outside the cluster, like `nuctl`, `curl` or any other HTTP client.
-
-To understand whether a function is reachable to your HTTP client (this can be `curl`, `httpie`, `nuctl invoke` or any
-other HTTP client), consider where it is running from and the network path. An unexposed (default) function will not be
-reachable from your client unless you're running the client from inside a pod in your cluster.
-
-If you wish to expose your function externally, for example, to be able to run `nuctl invoke` from outside the
-Kubernetes network, you can do so in one of 2 ways during deployment, both controlled via the [HTTP trigger spec](/docs/reference/triggers/http.md):
-1. Configure the function with a reachable [HTTP ingress](/docs/reference/triggers/http.md#attributes-ingresses). For
-   this to work you'll need to install an ingress controller on your cluster. See [function ingress document](/docs/concepts/k8s/function-ingress.md)
-   for more details.
-2. Configure the function to use [serviceType](/docs/reference/triggers/http.md#attributes-serviceType) of type `nodePort`.
-
-If you are deploying the function using [nuctl](/docs/reference/nuctl/nuctl.md) CLI, you can also configure a `nodePort` easily by using the
-`--http-trigger-service-type=nodePort` CLI arg.
+> **Security Note:** Exposing your functions outside of your Kubernetes cluster network has significant security implications.
+> Make sure that you understand the risks involved before deciding to expose any function externally.
+> Always control on which networks your functions are exposed and use proper authentication to protect these networks and the rest of your pipeline and data.
+
+By default, when deploying a function on a Kubernetes cluster the function is exposed only on the Kubernetes-cluster network, using a `ClusterIP` [Service Type](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types), and isn't exposed for external communication.
+In most network topologies, this makes the function available only inside the cluster network, and [Kubernetes network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) can be used to further limit and control communications.
+The function will be unavailable for invocations over HTTP from entities running outside of the cluster - such as `nuctl`, `curl`, or any other HTTP client.
+
+To understand whether a function can be reached by your HTTP client (which can be `curl`, `httpie`, `nuctl invoke`, or any other HTTP client), consider the location from which the function is running and the network path.
+An unexposed (default) function won't be reachable by your client unless you're running the client from inside a pod in your Kubernetes cluster.
```

Comment:
```suggestion
An unexposed (default) function won't be reachable from your client unless you're running the client from inside a pod in your Kubernetes cluster.
```
More accurate in this context.
The emphasis here is that it's not that the "client can't reach the function" because the client is at fault, but because, it's unreachable from where the client is, by nature of the network topology

Reply:
CHANGED (commit e5098ab)